libfreerdp/codec/planar.c

```diff
@@ -223,12 +223,57 @@ static INT32 planar_decompress_plane_rle(const BYTE* pSrcData, UINT32 SrcSize,
 	return (INT32)(srcp - pSrcData);
 }
 
+static BOOL writeLine(BYTE** ppRgba, UINT32 DstFormat, UINT32 width, const BYTE** ppR,
+                      const BYTE** ppG, const BYTE** ppB, const BYTE** ppA)
+{
+	UINT32 x;
+
+	if (!ppRgba || !ppR || !ppG || !ppB)
+		return FALSE;
+
+	switch (DstFormat)
+	{
+		case PIXEL_FORMAT_BGRA32:
+			for (x = 0; x < width; x++)
+			{
+				*(*ppRgba)++ = *(*ppB)++;
+				*(*ppRgba)++ = *(*ppG)++;
+				*(*ppRgba)++ = *(*ppR)++;
+				*(*ppRgba)++ = *(*ppA)++;
+			}
+
+			return TRUE;
+
+		case PIXEL_FORMAT_BGRX32:
+			for (x = 0; x < width; x++)
+			{
+				*(*ppRgba)++ = *(*ppB)++;
+				*(*ppRgba)++ = *(*ppG)++;
+				*(*ppRgba)++ = *(*ppR)++;
+				*(*ppRgba)++ = *(*ppA)++;
+			}
+
+			return TRUE;
+
+		default:
+			for (x = 0; x < width; x++)
+			{
+				BYTE alpha = (ppA) ? *(*ppA)++ : 0xFF;
```

Comment:
Could improve performance a bit by making two for-loops - one for alpha and one for non-alpha formats